contracts/rateproducer/rateproducer.cpp

```diff
@@ -10,108 +10,177 @@
+      }
 
+     
       // the payload must be ratings_json.
       check(ratings_json[0] == '{', "payload must be ratings_json");
       check(ratings_json[ratings_json.size()-1] == '}', "payload must be ratings_json");
+      
+      bp_rate_stats bp_stats = {0,0,0,0,0};
+      bool flag =  process_json_stats( ratings_json,&bp_stats);
+      if(flag){
+        // upsert bp rating
+        producers_table bps(_self, _self.value);
+        auto uniq_rating = (static_cast<uint128_t>(user.value) << 64) | bp.value;
 
-      // upsert bp rating
-      producers_table bps(_self, _self.value);
-      auto uniq_rating = (static_cast<uint128_t>(user.value) << 64) | bp.value;
-
-      auto uniq_rating_index = bps.get_index<name("uniqrating")>();
-      auto existing_rating = uniq_rating_index.find(uniq_rating);
+        auto uniq_rating_index = bps.get_index<name("uniqrating")>();
+        auto existing_rating = uniq_rating_index.find(uniq_rating);
 
-      uint64_t now = eosio::current_time_point().time_since_epoch().count();
+        uint64_t now = eosio::current_time_point().time_since_epoch().count();
 
-      if( existing_rating == uniq_rating_index.end() ) {
+        if( existing_rating == uniq_rating_index.end() ) {
           bps.emplace(_self, [&]( auto& row ) {
             row.id = bps.available_primary_key();
             row.uniq_rating = uniq_rating;
             row.user = user;
             row.bp = bp;
             row.created_at = now;
             row.updated_at = now;
-            row.ratings_json = ratings_json;
+            row.transparency = bp_stats.transparency;
+            row.infrastructure = bp_stats.infrastructure;
+            row.trustiness = bp_stats.trustiness;
+            row.community = bp_stats.community;
+            row.development = bp_stats.development ;   
           });
-          //update the general data
-         process_json_stats(bp,ratings_json);
-
-      } else {
-         uniq_rating_index.modify(existing_rating, _self, [&]( auto& row ) {
-           row.user = user;
-           row.bp = bp;
-           row.updated_at = now;
-           row.ratings_json = ratings_json;
-         });
-         //update the general data
-        process_json_stats(bp,ratings_json);
-
-       }
+          
+        } else {
+          uniq_rating_index.modify(existing_rating, _self, [&]( auto& row ) {
+            row.user = user;
+            row.bp = bp;
+            row.updated_at = now;
+            row.transparency = bp_stats.transparency;
+            row.infrastructure = bp_stats.infrastructure;
+            row.trustiness = bp_stats.trustiness;
+            row.community = bp_stats.community;
+            row.development = bp_stats.development ;  
+          });
+          
+        }
+        //update stats
+        save_bp_stats(bp,&bp_stats);
+      }
+      
     }
 
-    void process_json_stats(name bp_name,string ratings_json){
+    bool process_json_stats(string ratings_json,bp_rate_stats * a_bp_stats){
```

Comment:
Just a comment, if we are no using a json, why the input is a json and not just a float values?

Reply:
@rubenabix  theres a explanation within issue #257 's thread